src/ngraph/op/fused/lstm_cell.cpp

```diff
@@ -0,0 +1,352 @@
+               vector<float>{},
+               0.f,
+               false)
+{
+}
+
+op::LSTMCell::LSTMCell(const shared_ptr<Node>& X,
+                       const shared_ptr<Node>& W,
+                       const shared_ptr<Node>& R,
+                       const shared_ptr<Node>& H_t,
+                       const shared_ptr<Node>& C_t,
+                       size_t hidden_size,
+                       const vector<string>& activations,
+                       const vector<float>& activation_alpha,
+                       const vector<float>& activation_beta,
+                       float clip,
+                       bool input_forget)
+    : FusedOp("LSTMCell", {X, W, R, H_t, C_t})
+    , RNNCellBase(hidden_size, clip, activations, activation_alpha, activation_beta)
+    , m_X{X}
+    , m_W{W}
+    , m_R{R}
+    , m_H_t{H_t}
+    , m_C_t{C_t}
+    , m_activation_f{get_activation_function(0)}
+    , m_activation_g{get_activation_function(1)}
+    , m_activation_h{get_activation_function(2)}
+    , m_input_forget{input_forget}
+{
+    // Normally we would split B onto Wb an Rb and add them, however here they are all zeros,
+    // thus just initialize bias with appropriate shape and zeros.
+    m_bias = ngraph::op::Constant::create(element::f32,
+                                          Shape{m_gates_count * get_hidden_size()},
+                                          vector<float>(m_gates_count * get_hidden_size(), 0.f));
+
+    const auto& peephole_weights =
+        ngraph::op::Constant::create(element::f32,
+                                     Shape{m_peepholes_count * get_hidden_size()},
+                                     vector<float>(m_peepholes_count * get_hidden_size(), 0.f));
+    m_p_iof = builder::split(peephole_weights, m_peepholes_count);
+    constructor_validate_and_infer_types();
+}
+
+op::LSTMCell::LSTMCell(const shared_ptr<Node>& X,
+                       const shared_ptr<Node>& W,
+                       const shared_ptr<Node>& R,
+                       const shared_ptr<Node>& H_t,
+                       const shared_ptr<Node>& C_t,
+                       size_t hidden_size,
+                       const shared_ptr<Node>& B,
+                       const shared_ptr<Node>& P,
+                       const vector<string>& activations,
+                       const vector<float>& activation_alpha,
+                       const vector<float>& activation_beta,
+                       float clip,
+                       bool input_forget)
+    : FusedOp("LSTMCell", {X, W, R, H_t, C_t, B, P})
+    , RNNCellBase(hidden_size, clip, activations, activation_alpha, activation_beta)
+    , m_X{X}
+    , m_W{W}
+    , m_R{R}
+    , m_H_t{H_t}
+    , m_C_t{C_t}
+    , m_activation_f{get_activation_function(0)}
+    , m_activation_g{get_activation_function(1)}
+    , m_activation_h{get_activation_function(2)}
+    , m_input_forget{input_forget}
+{
+    // Split B onto Wb and Rb and add them.
+    NODE_VALIDATION_CHECK(this,
```

Comment:
All checks should be done in `pre_validate_and_infer_types()`

Reply:
Unfortunately I couldn't do that this time. I'm checking here since I'm using this input to produce other value. However this is not the biggest problem. But in lines https://github.com/NervanaSystems/ngraph/pull/2966/files#diff-cebfe1ce12ae7674ed2d17869b5893beR171 and https://github.com/NervanaSystems/ngraph/pull/2966/files#diff-cebfe1ce12ae7674ed2d17869b5893beR181  I have to check whether I have valid data and generate those values and couldn't do that after `constructor_validate_and_infer_types`, since it calls `decompose_op` which would try to use uninitialized values otherwise.

Reply:
Other solution would be to move those lines (171, 181) directly to `decompose_op`, but thought that it is cleaner to initialize those members in constructor...

Reply:
I think we probably will need to do it in `decompose_op`, unfortunately. We're moving towards taking validation checks out of constructors, so that revalidation after passes can be done.